File: app/src/main/java/knf/kuma/random/RandomAdapter.java
package knf.kuma.random;

import android.app.Activity;
import android.preference.PreferenceManager;
import android.support.annotation.LayoutRes;
import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import knf.kuma.R;
import knf.kuma.animeinfo.ActivityAnime;
import knf.kuma.commons.PatternUtil;
import knf.kuma.commons.PicassoSingle;
import knf.kuma.pojos.AnimeObject;

public class RandomAdapter extends RecyclerView.Adapter<RandomAdapter.RandomItem> {

    private Activity activity;
    private List<AnimeObject> list = new ArrayList<>();

    public RandomAdapter(Activity activity) {
        this.activity = activity;
    }

    @Override
    public RandomItem onCreateViewHolder(ViewGroup parent, int viewType) {
        return new RandomItem(LayoutInflater.from(activity).inflate(getLayout(), parent, false));
    }

    @LayoutRes
    private int getLayout() {
        if (PreferenceManager.getDefaultSharedPreferences(activity).getString("lay_type", "0").equals("0")) {
            return R.layout.item_fav;
        } else {
            return R.layout.item_fav_grid;
        }
    }

    @Override
    public void onBindViewHolder(final RandomItem holder, int position) {
        final AnimeObject animeObject = list.get(position);
        PicassoSingle.get(activity).load(PatternUtil.getCover(animeObject.aid)).into(holder.imageView);
        holder.title.setText(animeObject.name);
        holder.type.setText(animeObject.type);
        holder.cardView.setOnClickListener(v -> ActivityAnime.open(activity, animeObject, holder.imageView, false, true));
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public void update(List<AnimeObject> list) {
        this.list = list;
        notifyDataSetChanged();
    }

    class RandomItem extends RecyclerView.ViewHolder {
        @BindView(R.id.card)
        CardView cardView;
        @BindView(R.id.img)
        ImageView imageView;
        @BindView(R.id.title)
        TextView title;
        @BindView(R.id.type)
        TextView type;

        public RandomItem(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/knf/kuma/random/RandomAdapter.kt
package knf.kuma.random

import android.app.Activity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.annotation.LayoutRes
import androidx.recyclerview.widget.RecyclerView
import knf.kuma.R
import knf.kuma.animeinfo.ActivityAnime
import knf.kuma.commons.DesignUtils
import knf.kuma.commons.PrefsUtil
import knf.kuma.commons.bind
import knf.kuma.commons.load
import knf.kuma.pojos.av1.DirectoryAV1Min

internal class RandomAdapter(private val activity: Activity) : RecyclerView.Adapter<RandomAdapter.RandomItem>() {
    private var list: List<DirectoryAV1Min> = ArrayList()

    private val layout: Int
        @LayoutRes
        get() = if (PrefsUtil.layType == "0") {
            if (DesignUtils.isFlat) R.layout.item_fav_material else R.layout.item_fav
        } else {
            if (DesignUtils.isFlat) R.layout.item_fav_grid_material else R.layout.item_fav_grid
        }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RandomItem {
        return RandomItem(LayoutInflater.from(activity).inflate(layout, parent, false))
    }

    override fun onBindViewHolder(holder: RandomItem, position: Int) {
        val animeObject = list[position]
        holder.imageView.load(animeObject.imageUrl)
        holder.title.text = animeObject.name
        holder.type.text = getType(animeObject.category)
        holder.cardView.setOnClickListener { ActivityAnime.open(activity, animeObject, holder.imageView, true, true) }
    }

    override fun getItemCount(): Int {
        return list.size
    }

    fun getType(category: Int): String {
        return when (category) {
            1 -> "Anime"
            2 -> "Película"
            3 -> "OVA"
            else -> "Especial"
        }
    }

    fun update(list: List<DirectoryAV1Min>) {
        this.list = list
        notifyDataSetChanged()
    }

    internal class RandomItem(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val cardView: View by itemView.bind(R.id.card)
        val imageView: ImageView by itemView.bind(R.id.img)
        val title: TextView by itemView.bind(R.id.title)
        val type: TextView by itemView.bind(R.id.type)
    }
}
